File: app/src/main/java/com/kaciula/archiman/presentation/screen/userdetails/UserDetailsComponent.java
package com.kaciula.archiman.presentation.screen.userdetails;

import com.kaciula.archiman.injection.ActivityScope;
import com.kaciula.archiman.injection.AppComponent;
import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = UserDetailsModule.class)
interface UserDetailsComponent {

  void inject(UserDetailsController userDetailsController);
}

File: app/src/main/kotlin/com/kaciula/archiman/presentation/screens/userdetails/UserDetailsComponent.kt
package com.kaciula.archiman.presentation.screens.userdetails

import com.kaciula.archiman.injection.ActivityScope
import com.kaciula.archiman.injection.AppComponent
import dagger.Component

@ActivityScope
@Component(dependencies = [AppComponent::class], modules = [UserDetailsModule::class])
interface UserDetailsComponent {

    fun inject(userDetailsController: UserDetailsController)
}
